src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.MetadataDb.cs

```diff
@@ -165,28 +155,55 @@ public void Dispose()
                 Length = 0;
             }
 
-            internal void TrimExcess()
+            /// <summary>
+            /// If using array pools, trim excess if necessary.
+            /// If not using array pools, release the temporary array pool and alloc.
+            /// </summary>
+            internal void CompleteAllocations()
             {
-                // There's a chance that the size we have is the size we'd get for this
-                // amount of usage (particularly if Enlarge ever got called); and there's
-                // the small copy-cost associated with trimming anyways. "Is half-empty" is
-                // just a rough metric for "is trimming worth it?".
-                if (Length <= _data.Length / 2)
+                if (!_isLocked)
                 {
-                    byte[] newRent = ArrayPool<byte>.Shared.Rent(Length);
-                    byte[] returnBuf = newRent;
-
-                    if (newRent.Length < _data.Length)
+                    if (_convertToAlloc)
                     {
-                        Buffer.BlockCopy(_data, 0, newRent, 0, Length);
-                        returnBuf = _data;
-                        _data = newRent;
+                        // Mutate the _data. This is faster and allocates less than creating a new MetadataDb.
+                        // This is used by JsonElement.Parse() so there are no concurrency concerns since the element
+                        // is not returned until it is done parsing.
```

Comment:
I'm not sure the comment really makes sense here.  The else block also replaces `_data`, and nothing makes a new MetadataDb.

I'd just delete this comment block altogether.